sentry/src/routes/campaign.rs

```diff
@@ -17,28 +38,320 @@ pub async fn create_campaign<A: Adapter>(
 
     // TODO AIP#61: Validate Campaign
 
-    let error_response = ResponseError::BadRequest("err occurred; please try again later".into());
+    let error_response = ResponseError::BadRequest("err occurred; please try again later".to_string());
+
+    let accounting_spent = get_accounting_spent(app.pool.clone(), &campaign.creator, &campaign.channel.id()).await?;
+
+    // TODO: AIP#61: Update when changes to Spendable are ready
+    let latest_spendable = fetch_spendable(app.pool.clone(), &campaign.creator, &campaign.channel.id()).await?;
+    let remaining_for_channel = get_total_remaining_for_channel(&accounting_spent, &latest_spendable)?;
+
+    if campaign.budget > remaining_for_channel {
+        return Err(ResponseError::Conflict("Not Enough budget for campaign".to_string()));
+    }
+
+    // If the channel is being created, the amount spent is 0, therefore remaining = budget
+    update_remaining_for_campaign(&app.redis.clone(), campaign.id, campaign.budget).await?;
 
     // insert Campaign
-    
-    // match insert_campaign(&app.pool, &campaign).await {
-    //     Err(error) => {
-    //         error!(&app.logger, "{}", &error; "module" => "create_channel");
-
-    //         match error {
-    //             PoolError::Backend(error) if error.code() == Some(&SqlState::UNIQUE_VIOLATION) => {
-    //                 Err(ResponseError::Conflict(
-    //                     "channel already exists".to_string(),
-    //                 ))
-    //             }
-    //             _ => Err(error_response),
-    //         }
-    //     }
-    //     Ok(false) => Err(error_response),
-    //     _ => Ok(()),
-    // }?;
-
-    let create_response = SuccessResponse { success: true };
+    match insert_campaign(&app.pool, &campaign).await {
+        Err(error) => {
+            error!(&app.logger, "{}", &error; "module" => "create_campaign");
+            return Err(ResponseError::Conflict("campaign already exists".to_string()));
```

Comment:
Only a single `error.code()` means that this campaign already exist:

`PoolError::Backend(error) if error.code() = SqlState::UNIQUE_VIOLATION`

The rest of the errors should be handled with `BadRequest` & the `error.to_string()`

- [ ] check against UniqueViolation and return `Conflict` response
- [ ] `Ok(false)` - stays the same, just use `=> Err(ResponseError::BadRequest("Encountered error while creating Campaign; please try again".to_string()))`